ansible/roles/lib_git/build/lib/ssh_agent.py

```diff
@@ -0,0 +1,130 @@
+        """
+        def __init__(self, ssh_agent, ssh_auth_sock_dir):
+            self.ssh_agent = ssh_agent
+            self.ssh_auth_sock_dir = ssh_auth_sock_dir
+            self.cleaned_up = False
+            self.original_env_var = os.environ.get('SSH_AUTH_SOCK')
+        def __call__(self):
+            if self.cleaned_up:
+                return
+            self.cleaned_up = True
+            try:
+                shutil.rmtree(self.ssh_auth_sock_dir, ignore_errors=True)
+            except OSError:
+                pass
+            try:
+                self.ssh_agent.kill()
+            except OSError:
+                pass
+            if self.original_env_var:
+                os.environ['SSH_AUTH_SOCK'] = self.original_env_var
+            else:
+                del os.environ['SSH_AUTH_SOCK']
+        def pass_(self):
+            """A function to appease pylint"""
+            pass
+        def pass__(self):
+            """Another function to appease pylint"""
+            self.pass_()
+    def __init__(self):
+        devnull = open(os.devnull, 'w')
+        # Start an ssh-agent process and register it to be killed atexit
+        self.ssh_auth_sock_dir = tempfile.mkdtemp(prefix=os.path.basename(sys.argv[0]) + '.')
+        self.ssh_auth_sock = os.path.join(self.ssh_auth_sock_dir, "ssh_agent")
+        self.ssh_agent = subprocess.Popen(["ssh-agent", "-d", "-a", self.ssh_auth_sock], stdout=devnull, stderr=devnull)
+        self.cleanup = self.Cleanup(self.ssh_agent, self.ssh_auth_sock_dir)
+        # this is here so that when python exits, we make sure that the agent is killed
+        # (in case python exits before our __del__() is called
+        atexit.register(self.cleanup)
+        os.environ["SSH_AUTH_SOCK"] = self.ssh_auth_sock
+    def __enter__(self):
+        return self
+    def __exit__(self, exc_type, exc_value, tback):
+        self.cleanup()
+    def __del__(self):
+        self.cleanup()
+
+    def kill(self):
+        '''Explicitly kill the running ssh-agent
+
+        It's not necessary to call this function as the agent
+        will be cleaned up automatically.
+        '''
+        self.cleanup()
+
+    def add_key(self, key):
+        """Add a key to the running agent.
+
+        Note:
+            This function can be called any number of times to add multiple keys.
+
+        Args:
+            key (str): A string containing the ssh private key to be added (the
+                    actual key data, not the filename of a key)
+        Raises:
+            SshAgentException: when ssh-add does not immediately return (as in the
+                    case of a private key with a passphrase)
+        """
+        #if self.ssh_agent.poll() is None:
+        #    raise SshAgentException("Unable to add ssh key. Did agent die?")
+        named_pipe_path = os.path.join(self.ssh_auth_sock_dir, "keypipe." + str(random.getrandbits(64)))
+        try:
+            os.mkfifo(named_pipe_path, 0600)
```

Comment:
no, python has the same number formatting prefixes as C and many languages since. numbers starting with 0 are octal, just like numbers starting with 0x are hexidecimal.